aspnetcore/performance/caching/middleware.md

```diff
@@ -1,48 +1,65 @@
+This document provides details on how to configure the Response Caching Middleware in ASP.NET Core apps. The middleware determines when responses are cacheable, stores responses, and serves responses from cache. For an introduction to HTTP caching and the `ResponseCache` attribute, see [Response Caching](response.md).
 
 ## Package
-To include the middleware in your project, add a reference to the  [`Microsoft.AspNetCore.ResponseCaching`](https://www.nuget.org/packages/Microsoft.AspNetCore.ResponseCaching/) package. The middleware depends on .NET Framework 4.5.1 or .NET Standard 1.3 or later. This feature is available for apps that target ASP.NET Core 1.1.0 or later.
+To include the middleware in a project, add a reference to the  [`Microsoft.AspNetCore.ResponseCaching`](https://www.nuget.org/packages/Microsoft.AspNetCore.ResponseCaching/) package. This feature is available for apps that target ASP.NET Core 1.1 or later.
 
 ## Configuration
-In `ConfigureServices`, add the middleware to your service collection.
+In `ConfigureServices`, add the middleware to the service collection.
 
-[!code-csharp[Main](middleware/sample/Startup.cs?name=snippet1)]
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-Configure the application to use the middleware when processing requests. The sample application adds a [`Cache-Control`](https://tools.ietf.org/html/rfc7234#section-5.2) header to the response that will cache cacheable responses for up to 10 seconds. The sample also sends a [`Vary`](https://tools.ietf.org/html/rfc7231#section-7.1.4) header to configure the cache to serve the response only if the [`Accept-Encoding`](https://tools.ietf.org/html/rfc7231#section-5.3.4) header of subsequent requests matches that from the original request.
+[!code-csharp[Main](middleware/samples/2.x/Program.cs?name=snippet1&highlight=4)]
 
-[!code-csharp[Main](middleware/sample/Startup.cs?name=snippet2)]
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
-The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
+[!code-csharp[Main](middleware/samples/1.x/Startup.cs?name=snippet1&highlight=3)]
 
->[!WARNING]
+---
+
+Configure the app to use the middleware when processing requests. The sample app adds a [`Cache-Control`](https://tools.ietf.org/html/rfc7234#section-5.2) header to the response that caches cacheable responses for up to 10 seconds. The sample sends a [`Vary`](https://tools.ietf.org/html/rfc7231#section-7.1.4) header to configure the middleware to serve a cached response only if the [`Accept-Encoding`](https://tools.ietf.org/html/rfc7231#section-5.3.4) header of subsequent requests matches that of the original request.
```

Comment:
> this command

If you mean the `Cache-Control` header bit, I wonder if this should just refer back to the topic we hope they read first it's mentioned in the opening para): https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response. I can add a quick statement to that effect.

Otherwise, let me know which bit needs more work ... the `UseResponseCaching`, the `app.Run`, the `Cache-Control`, or the `WriteAsync` bit, and I'll add more text here.